DOCS-12178-whats-new-4.4-cv (#2076)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/whats-new-in-mule.adoc

= What's New in Mule 4.4
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> Mule runtime engine 4.4 is mainly focused on fixing issues, and also includes some enhancements.

Reviewer: ```suggestion
Mule runtime engine 4.4 fixes issues and includes some enhancements.
```

> Mule runtime engine 4.4 fixes issues and includes some enhancements. Deploy all your new and existing applications to Mule 4.4 to benefit from all the improvements in this version.

Reviewer: ```suggestion
Mule runtime engine 4.4 provides important enhancements and fixes. Deploy all your new and existing applications to Mule 4.4 to benefit from all the improvements in this version.
```

== Feature Flags

> Mule 4.4 incorporates a feature flagging mechanism, which enables you to disable certain features that change core functionality in Mule to avoid backward compatibility issues.

Reviewer: These are disabled by default. People might think they have to disable them. The feature flag doc makes that clear though.

Reviewer: ```suggestion
Mule 4.4 incorporates a feature flagging mechanism, which enables you to control the enablement of certain Mule features that affect backward compatibility.
```

See xref:mule-runtime::feature-flagging.adoc[Feature Flagging] for additional information and configuration instructions.

== Correlation ID Management
Mule 4.4 enables you to modify the Correlation ID associated with a Mule event. You can also change the algorithm Mule uses to generate each Correlation ID.

See xref:mule-runtime::correlation-id.adoc[Correlation ID] for additional information and configuration instructions.

== Improved Logging Capabilities
Mule 4.4 incorporates Mapped Diagnostic Context (MDC), which enriches logging and improves tracking by providing more context or information in the logs for Mule events.

By using the Mule Tracing module, you can add, remove, and clear variables from the logging context for a given Mule event.

See xref:mule-runtime::logging-mdc.adoc[MDC Logging] for additional information and configuration instructions.

== DataWeave Updates

The Mule 4.4 release includes DataWeave 2.4.0 features.

See xref:2.4@dataweave::whats-new-in-dw.adoc[What's New in DataWeave] for additional information.

== Core Components

> Mule 4.4 introduces the Mule Tracing Module (available in Studio 7.11), which enables correlation ID management capabilities as well as improved logging options.

Reviewer: ```suggestion
Mule 4.4 introduces the Mule Tracing Module (available in Studio 7.11), which enables Correlation ID management capabilities, as well as improved logging options.
```

Reviewer: matching "Correlation ID" mentioned previously

= What's New in Mule 4.4
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

> Mule 4.4 is the latest and best Mule Runtime available, it continues to deliver high performance, expedites application development on Anypoint Platform, and offers a resilient solution for use cases that are vital to the functioning of your organization.

Reviewer: This is marketing language not suitable for technical documentation.
